azure/datalake/store/core.py

```diff
@@ -192,6 +197,154 @@ def chmod(self, path, mod):
         path = AzureDLPath(path).trim()
         self.azure.call('SETPERMISSION', path.as_posix(), permission=mod)
         self.invalidate_cache(path.as_posix())
+    
+    def set_expiry(self, path, expiry_option, expire_time=None):
+        """
+        Sets or removes the expiration time on the specified file. 
+        This operation can only be executed against files. 
+        
+        Note: Folders are not supported.
+
+        Parameters
+        ----------
+        path: str
+            File path to set or remove expiration time
+        expire_time: int
+            The time that the file will expire, corresponding to the expiry_option that was set
+        expiry_option: str
+            Indicates the type of expiration to use for the file: 
+                1. NeverExpire: ExpireTime is ignored. 
+                2. RelativeToNow: ExpireTime is an integer in milliseconds representing the expiration date relative to when file expiration is updated. 
+                3. RelativeToCreationDate: ExpireTime is an integer in milliseconds representing the expiration date relative to file creation. 
+                4. Absolute: ExpireTime is an integer in milliseconds, as a Unix timestamp relative to 1/1/1970 00:00:00.
+        """
+        parms = {}
+        valid_expire_types = [x.value for x in ExpiryOptionType]
+        value_to_use = [x for x in valid_expire_types if x.lower() == expiry_option.lower()]
```

Comment:
@brettcannon is there something like a "select" that I can do so it comes back as just one element? It will always be just one element so I would like to be able to just get that.